### DISCUSSION TITLE
RichTextEditor on a custom dropdownlist component

### DISCUSSION THREAD

**Discussion by zixiang2018** [GrapesJS/grapesjs#4888] [Q&A]
Hi, I am currently working on a WYSIWYG project and happened to chance upon this repo. Have been working on this for a few weeks now, but I have something that I want to customize on my canvas components.

I understand that the RichTextEditor(RTE) is for the "Text" component, but is it possible to display and use the RTE on a custom dropdownlist/select component? In my custom dropdownlist/select component, I plan to use the RTE for justifying the contents, font changes, etc like how the RTE would work on the "Text" component

Have been doing my research but couldnt find any results online. Looking forward to an answer from the community here! 🙇‍♂️

**Comment by artf** [comment]
Hi @zixiang2018 I think you should be able to reimplement RTE functionality by looking at [ComponentTextView](https://github.com/GrapesJS/grapesjs/blob/a1b54fc40659f00b2a673c10d673934c8067eef8/src/dom_components/view/ComponentTextView.ts) and doing similar in your custom dropdownlist/select component

**Reply by zixiang2018** [reply]
I've tried tinkering around with the rte.enable() function, but it does not seem to work as I am unsure where to get a few variables from.

Got a few questions,

1. [Where can I get this `em`](https://github.com/GrapesJS/grapesjs/blob/a1b54fc40659f00b2a673c10d673934c8067eef8/src/dom_components/view/ComponentTextView.ts#L28) (I believe it is the EditorModule) from my custom component (the `editor.DomComponents.addType`)
2. How do I enable the RTE to open up on selection of my component (any code references)? Is there a way to not import the RichTextEditor and RichTextEditorModule from [here](https://github.com/GrapesJS/grapesjs/blob/a1b54fc40659f00b2a673c10d673934c8067eef8/src/dom_components/view/ComponentTextView.ts#L3-L4) (I cant seem to import it into my project)?

**Reply by artf** [reply]
When you define your custom component view the `em` it's already available in the view context and you don't need to import RichTextEditor stuff (those are here only for TS definitions), you can also get RTE instance from `editor`
```js
editor.Components.addType('...', {
// ...
  view: {
  // ...
    async onActive(ev: Event) {
      const { em } = this;
     // ... use the editor to get RTE
     editor.RichTextEditor.enable(this)
     // ... 
    }
  }
})
```
